From 8ae0f886b7abded65a0268a0496ff6e016a5e868 Mon Sep 17 00:00:00 2001
From: Paul Gortmaker <paul.gortmaker@windriver.com>
Date: Mon, 3 Apr 2017 22:58:39 -0400
Subject: [PATCH] vim: Make text mode editing Great again.

Some of the default vim settings can be downright annoying, and are not
enabled on most standard distros out there.

The "bad" settings are"

	set mouse=a
		-breaks basic cut-and-paste

	set autoindent
		-breaks paste of a code block

	set backup
		-pollutes "git status" with lots of "~" files

	set incsearch
		-causes source window to twitch and flail while typing

	jump-to-last-position
		-fails miserably to be aware of temp files.  The temporary
		 git commit log buffer always has the same file name.

I've added comments to clearly indicate this change does deviate from the
vim default, so offended people can direct their complaints accordingly.

Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>

diff --git a/runtime/defaults.vim b/runtime/defaults.vim
index 43ff1e0fbed4..ea116965c308 100644
--- a/runtime/defaults.vim
+++ b/runtime/defaults.vim
@@ -3,6 +3,10 @@
 " Maintainer:	Bram Moolenaar <Bram@vim.org>
 " Last change:	2016 Sep 02
 "
+" Modified:	Paul Gortmaker <paul.gortmaker@windriver.com>
+" Reason:	better alignment with "typical" distro settings.
+" Last change:	2017 Apr 03
+"
 " This is loaded if no vimrc file was found.
 " Except when Vim is run with "-u NONE" or "-C".
 " Individual settings can be reverted with ":set option&".
@@ -39,12 +43,12 @@ set display=truncate
 
 " Show a few lines of context around the cursor.  Note that this makes the
 " text scroll if you mouse-click near the start or end of the window.
-set scrolloff=5
+" set scrolloff=5
 
 " Do incremental searching when it's possible to timeout.
-if has('reltime')
-  set incsearch
-endif
+"if has('reltime')
+"  set incsearch
+"endif
 
 " Do not recognize octal numbers for Ctrl-A and Ctrl-X, most users find it
 " confusing.
@@ -64,11 +68,12 @@ map Q gq
 " Revert with ":iunmap <C-U>".
 inoremap <C-U> <C-G>u<C-U>
 
-" In many terminal emulators the mouse works just fine.  By enabling it you
-" can position the cursor, Visually select and scroll with the mouse.
-if has('mouse')
-  set mouse=a
-endif
+" In some terminal emulators the mouse kind of works.  By enabling it you
+" can position the cursor, visually select and scroll with the mouse.
+" Note that it does break a very common paste use case if you enable it.
+" if has('mouse')
+"   set mouse=a
+" endif
 
 " Switch syntax highlighting on when the terminal has colors or when using the
 " GUI (which always has colors).
@@ -90,22 +95,6 @@ if has("autocmd")
   " Also load indent files, to automatically do language-dependent indenting.
   " Revert with ":filetype off".
   filetype plugin indent on
-
-  " Put these in an autocmd group, so that you can revert them with:
-  " ":augroup vimStartup | au! | augroup END"
-  augroup vimStartup
-    au!
-
-    " When editing a file, always jump to the last known cursor position.
-    " Don't do it when the position is invalid or when inside an event handler
-    " (happens when dropping a file on gvim).
-    autocmd BufReadPost *
-      \ if line("'\"") >= 1 && line("'\"") <= line("$") |
-      \   exe "normal! g`\"" |
-      \ endif
-
-  augroup END
-
 endif " has("autocmd")
 
 " Convenient command to see the difference between the current buffer and the
diff --git a/runtime/evim.vim b/runtime/evim.vim
index f80b3c03b627..c9de16371457 100644
--- a/runtime/evim.vim
+++ b/runtime/evim.vim
@@ -2,6 +2,10 @@
 " Maintainer:	Bram Moolenaar <Bram@vim.org>
 " Last Change:	2016 Jul 24
 
+" Modified:	Paul Gortmaker <paul.gortmaker@windriver.com>
+" Reason:	better alignment with "typical" distro settings.
+" Last change:	2017 Apr 03
+
 " Don't use Vi-compatible mode.
 set nocompatible
 
@@ -27,16 +31,12 @@ cnoremap <silent> <C-F> <C-C>:promptfind<CR>
 
 
 set backspace=2		" allow backspacing over everything in insert mode
-set autoindent		" always set autoindenting on
-if has("vms")
-  set nobackup		" do not keep a backup file, use versions instead
-else
-  set backup		" keep a backup file
-endif
+" set autoindent	" always set autoindenting on
+set nobackup		" do not keep a backup file
 set history=50		" keep 50 lines of command line history
 set ruler		" show the cursor position all the time
-set incsearch		" do incremental searching
-set mouse=a		" always use the mouse
+" set incsearch		" do incremental searching
+" set mouse=a		" always use the mouse
 
 " Don't use Ex mode, use Q for formatting
 map Q gq
-- 
2.7.4

